projects/plugins/jetpack/_inc/client/newsletter/subscriptions-settings.jsx

```diff
@@ -95,86 +98,123 @@ function SubscriptionsSettings( props ) {
 	const isDisabled =
 		! isSubscriptionsActive || unavailableInOfflineMode || isSavingAnyOption( [ 'subscriptions' ] );
 
+	const changeWelcomeMessageState = useCallback(
+		event => {
+			const subscriptionOptionEvent = {
+				target: { name: event.target.name, value: { welcome: event.target.value } },
+			};
+			onOptionChange( subscriptionOptionEvent );
```

Comment:
I see that overriding the existing `event.target.value` would actually impact the actual value on the textarea element hence a fake event object was created instead. Since `onOptionChange` is only expecting `event.target.name` and `event.target.value` I think this solution is OK.

I see that `value: { welcome: event.target.value }` is not inheriting the other properties in the existing object, e.g. `value: { ...oldSubscriptionOptions, welcome: event.target.value }`. This is not necessary at this point because the other two props `invitation` and `comment_follow` are now deprecated, and the backend is actually going an `array_merge` on the old and new subscription options (https://github.com/Automattic/jetpack/pull/35621/files#diff-f8e5ef1115599de750b64143dd1901554254eddd95ab4371b6b6b3b2a5914224R1029).

If in the future we're introducing new props in `subscription_options` that is also being modified on the client-side, it would be important to remember to take precaution that this part of the code isn't inheriting from the existing potentially updated subscription options state.